tests/tests_pytorch/test_cli.py

```diff
@@ -1860,6 +1860,34 @@ def test_lightning_cli_args_and_sys_argv_warning():
         LightningCLI(TestModel, run=False, args=["--model.foo=789"])
 
 
+def test_add_class_args_required_false_skips_addition(tmp_path):
+    from lightning.pytorch import callbacks, cli
+
+    class FooCheckpoint(callbacks.ModelCheckpoint):
+        def __init__(self, dirpath, *args, **kwargs):
+            super().__init__(dirpath, *args, **kwargs)
```

Comment:
```suggestion
```

There is no reason for this class since `ModelCheckpoint` could be used directly.